src/components/card/card.js

```diff
@@ -2,18 +2,26 @@ import { getMenu } from "../../api/getApi";
 import { addMenu } from "./createCards";
 
 export async function getCardsData(value) {
-    try {
-      const data = await getMenu();
 
-      const nameCategory = value.toLowerCase();
-      const arrCardsByCategory = data[0][nameCategory];
+  try {
+    const data = await getMenu();
 
-      arrCardsByCategory.forEach((obj) => addMenu(obj));
+    const nameCategory = value.toLowerCase();
+    const arrCardsByCategory = data[0][nameCategory];
 
-    } catch (e) {
-      console.error("error", e);
-      throw e;
-    }
+    deleteCardsByCategory();
+    arrCardsByCategory.forEach((obj) => addMenu(obj));
+
+  } catch (e) {
+    console.error("error", e);
+    throw e;
+  }
+}
+
+function deleteCardsByCategory() {
+  
+  const cards = document.querySelectorAll(".card");
+  cards.forEach(item => item.remove());
 }
```

Comment:
const container = document.querySelector('.cards__container')
  container.innerHTML = null

You could just write like that
that's all))